pkg/ui/workspaces/cluster-ui/src/statementsDiagnostics/activateStatementDiagnosticsModal.tsx

```diff
@@ -218,6 +232,51 @@ export const ActivateStatementDiagnosticsModal = React.forwardRef(
             </Button.Group>
           </Radio.Group>
           <Divider type="horizontal" />
+
+          <Radio.Group
+            value={filterPerPlanGist}
+            className={cx("diagnostic__plan-gist-group")}
+          >
+            <Button.Group className={cx("diagnostic__btn-group")}>
+              <Radio
+                value={false}
+                className={cx("diagnostic__radio-btn", "margin-bottom")}
+                onChange={() => setFilterPerPlanGist(false)}
+              >
+                For all plan gists
+              </Radio>
+              <br />
+              <Radio
+                value={true}
+                className={cx("diagnostic__radio-btn")}
+                onChange={() => setFilterPerPlanGist(true)}
+              >
+                For the following plan gist:
+                <div className={cx("diagnostic__plan-gist-container")}>
+                  <Select
+                    disabled={!filterPerPlanGist}
+                    value={selectedPlanGist}
+                    defaultValue={planGists ? planGists[0] : ""}
+                    onChange={(selected: string) =>
+                      setSelectedPlanGist(selected)
```

Comment:
You can skip creating the anon function here and just pass in `setSelectedPlanGist` since the params line up and there's no extra work.

Reply:
I was getting a lint error, because the default is int, so I need to create this way to get the parameter as string